panel/util.py

```diff
@@ -256,33 +264,36 @@ def datetime_as_utctimestamp(value):
     return value.replace(tzinfo=dt.timezone.utc).timestamp() * 1000
 
 
-def is_number(s):
+def is_number(s: Any) -> bool:
```

Comment:
It would be even better to make this a type guard `def is_number(s: Any) -> TypeGuard[float]` (needs `typing_extensions` for now). This way `s` will be known to be a `float` in `if is_number(s):` or after `assert is_number(s)`.